### DISCUSSION TITLE
Welcome to Quizee Discussions!

### DISCUSSION THREAD

**Discussion by nitesh2920** [nitesh2920/Quizee#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nitesh2920** [comment]
Any thoughts, ideas , improvement  of code or adding new functionality in this project can be discussed here.

**Reply by RohitRaj766** [reply]
Hi, @nitesh2920 
My self Rohit Raj I am beginner in mern dev. I think i can contribute to this project. Can I join too ?

**Reply by nitesh2920** [reply]
Rohit, you're welcome to join this project. However, please note that the project has not yet started. I assume you don't have any issues with that.

**Reply by RohitRaj766** [reply]
Any idea when it will start ?

**Reply by nitesh2920** [reply]
soon.

**Comment by RohitRaj766** [comment]
This project can have 2 parts namely.
1) client (The users)
2) admin pannel (the super-users or admins to manage the quizee from gui pannel)
@nitesh2920

**Comment by RohitRaj766** [comment]
To manage the flow and maintenance of the project effectively, I propose two main branches:

1. Main Branch (**main**):
   This branch represents the production-ready code.
   It should only contain stable and tested code.
   Changes from the `development` branch are merged into `main` when they are ready to be deployed to production.

2. Development Branch (**development**):
   This branch serves as the primary development and testing environment.
   New features and fixes are developed and tested here.
   Only after thorough testing and verification are changes merged into `main`.

To adhere to best practices, always create a new branch for each new task or feature. Once the work is complete and thoroughly tested, merge it into the `development` branch for further testing. Only when the changes are deemed production-ready should they be merged into the `main` branch for deployment.

This workflow ensures a clear separation between development and production code, allowing for thorough testing before updates are released to the live environment.

**What is your opinion on this flow ?**

**Reply by nitesh2920** [reply]
Yes, Absolutely right, Rohit
